rasa/core/test.py

```diff
@@ -577,7 +616,12 @@ async def _predict_tracker_actions(
     agent: "Agent",
     fail_on_prediction_errors: bool = False,
     use_e2e: bool = False,
-) -> Tuple[EvaluationStore, DialogueStateTracker, List[Dict[Text, Any]]]:
+) -> Tuple[
+    EvaluationStore,
+    DialogueStateTracker,
+    List[Dict[Text, Any]],
+    List[EntityEvaluationResult],
+]:
```

Comment:
Not really related to this PR but that looks a bit crazy — I would prefer to use some custom data structure to at least have some names for all the stuff that's part of the tuple now.

Reply:
I think this area of the code base would need some major refactoring for us to remove these kinds of return types.